web-rdfs/source/http_handlers.cc

```diff
@@ -104,34 +103,63 @@ void get_handler(std::shared_ptr<HttpsServer::Response> response,
   LOG(DEBUG) << "Type of Request " << typeOfRequest;
   LOG(DEBUG) << "Path " << path;
 
-  if (!typeOfRequest.compare("DELETE")) {
-    delete_file_handler(response, path);
-  } else if (!typeOfRequest.compare("OPEN")) {
+  if (!typeOfRequest.compare("OPEN")) {
     read_file_handler(response, path);
-  } else if (!typeOfRequest.compare("MKDIR")) {
-    mkdir_handler(response, path);
-  } else if (!typeOfRequest.find("RENAME")) {
-    std::string pathForRename = typeOfRequest.substr(6);
-    rename_file_handler(response, path, pathForRename);
   } else {
-    create_file_handler(response, request);
+    response->write(SimpleWeb::StatusCode::client_error_bad_request);
   }
 }
 
 void post_handler(std::shared_ptr<HttpsServer::Response> response,
                   std::shared_ptr<HttpsServer::Request> request) {
-  // TODO(security): invoke another handler depending on qs opcode.
-  create_file_handler(response, request);
+  // Do not support post requests right now
+  response->write(SimpleWeb::StatusCode::client_error_bad_request);
 }
 
 void put_handler(std::shared_ptr<HttpsServer::Response> response,
                  std::shared_ptr<HttpsServer::Request> request) {
-  // TODO(security): invoke another handler depending on qs opcode.
-  create_file_handler(response, request);
+  std::string baseUrl = "/webhdfs/v1";
```

Comment:
There's a lot of repeated logic here, can we pull this out into a common routine that parses the query string according to webhdfs api rules?